dsl_parser/functions.py

```diff
@@ -122,10 +125,15 @@ def _register_entry_point_functions():
 
 def is_function(value):
     """Does the value represent a template function call?"""
-    # functions use the syntax {function_name: args}, so let's look for
+    # functions use the syntax {function_name: args}, or
+    # {function_name: args, 'type': type_name} so let's look for
     # dicts of length 1 where the only key was registered as a function
-    return isinstance(value, dict) and len(value) == 1 \
-        and list(value.keys())[0] in TEMPLATE_FUNCTIONS
+    if not isinstance(value, dict):
+        return False
+    value_copy = copy.copy(value)
+    value_copy.pop('type', None)
+    return len(value_copy) == 1 \
+        and list(value_copy.keys())[0] in TEMPLATE_FUNCTIONS
```

Comment:
`is_function` is called A LOT, so let's optimize it: copying is going to be slow. I'm thinking something like this:
```python
def is_function(value):
    if not isinstance(value, dict):
        return False
    if len(value) != 1 and len(value) != 2:
        return False
    for k in value:
        if k == 'type' or k in TEMPLATE_FUNCTIONS:
            continue
        else:
            return False
    return True
```

also, in that one place, we do `is_function` and then fetches it; maybe we don't want an `is_function`, maybe we want a `get_function` instead: it could be similar to the above,
```python
def get_function(value):
    [...]
    func = None
    for k in value:
        if k == 'type':
            continue
        func = TEMPLATE_FUNCTIONS.get(k)
    return func
```
and then all call-sites of `if is_function()` can be replaced to `if get_function() is None`

Reply:
or rather return both k and v for get_function, because `parse` would need both